core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -656,4 +667,55 @@ public static short getDefaultReplication(String path, FileFactory.FileType file
       throws IOException {
     return getCarbonFile(path, fileType).getDefaultReplication(path);
   }
+
+  /**
+   * Write content into specified file path
+   * @param content content to write
+   * @param filePath file path to write
+   * @throws IOException if IO errors
+   */
+  public static void writeFile(String content, String filePath) throws IOException {
+    AtomicFileOperations fileWrite = AtomicFileOperationFactory.getAtomicFileOperations(filePath);
+    BufferedWriter brWriter = null;
+    DataOutputStream dataOutputStream = null;
+    try {
+      dataOutputStream = fileWrite.openForWrite(FileWriteOperation.OVERWRITE);
+      brWriter = new BufferedWriter(new OutputStreamWriter(dataOutputStream,
+          Charset.forName(CarbonCommonConstants.DEFAULT_CHARSET)));
+      brWriter.write(content);
+      brWriter.flush();
```

Comment:
flush() can be moved to finally block and can add a null check

Reply:
I removed this line and brWriter will be closed in final block